snap/squashfs2/directory.go

```diff
@@ -0,0 +1,98 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2014-2018 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * https://www.kernel.org/doc/html/v5.8/filesystems/squashfs.html
+ */
+
+package squashfs2
+
+import "fmt"
+
+func (d *directory) readDirectoryHeader() squashfs_dir_header {
```

Comment:
(nitpick) this is probably more something that @pedronis should comment on but it would feel more natural/go-ish to me if this file also contains the `type squashfs_dir_header struct`, that seems to be more ideomatic than having it in a separate file.

Reply:
Thanks for all your comments Michael, they are greatly appreciated. Ive restructured the code a bit, but the point may still be valid here. I wanted to gather all the structures that are present in the image under the same namespace so they don't coincide with any structures that I define to keep state.